### PR TITLE
Knexify

### PR DESCRIPTION
**Pull request by tschuy**
This knexifies the project. You now run migrations with `npm run migrations`, and it uses knex internally to talk to the database. This allows you to change things from

``` javascript
// save a row to the db for each status message in a standup
function saveStatusRows(lastID, locals, callback) {
  var now = new Date().getTime(),
      statements = [];
  for (var k in STATES) {
    var key = STATES[k];
    for (var i=0; i<locals[key].length; i++) {
      if (locals[key][i].length) {
        statements.push(['INSERT INTO statuses VALUES (NULL, ?, ?, ?, ?, ?)',
                        [locals.irc_nick, now, k, locals[key][i], lastID]]);
      }
    }
  }

  async.forEach(statements,
      function(item, callback) {
        var stmt = db.prepare(item[0]);
        var args = item[1];
        stmt.run.apply(stmt, args);
        stmt.finalize(callback);
      },
      function(err) {
        callback(err);
      }
  );
}
```

to 

```
// save a row to the db for each status message in a standup
function saveStatusRows(lastID, locals, callback) {
  var now = new Date().getTime();
  statuses = [];
  for (var state of STATES) {
    // for every state, a user can submit multiple tasks with linebreaks
    for (var task of locals[state]) {
      statuses.push({name: locals.irc_nick, time: now, state: state, status: task, stats: lastID});
    }
  }

  knex('statuses').insert(statuses).then(function() {
    callback();
  }).catch(function(err) {
    callback(err);
  });
}
```

### CODE REVIEW

File: standupbot.js [hunk 1]
```diff
@@ -89,13 +52,15 @@ app.get('/', function(req, res) {
...
-    getStatusForID(req.cookies.lastID, function(err, result) {
+    knex('statuses').where({stats: req.cookies.lastID}).then(function(result) {
       for (var i=0; i<result.length; i++) {
         var row = result[i],
             key = STATES[row.state];
         locals[key].push(row.status);
       }
       render(locals);
+    }).catch(function(err) {
+      // TODO handle this error
```
Comment (iankronquist) [inline standupbot.js]: You should probably handle this error.

Maybe try:
`res.send(JSON.stringify({error: 'Database error', error_contents: err}));`

File: package.json [hunk 2]
```diff
@@ -25,6 +26,7 @@
   "scripts": {
     "start": "while(true) do node standupbot.js; sleep 5; done",
     "test": "mocha",
+    "migrations": "knex migrate:latest",
     "lint": "eslint ."
   },
   "engine": "node >= 0.8.14"
```
Comment (iankronquist) [inline package.json:32]: Do you think we should update the engine requirements and start using ES6? That doesn't belong in this PR, but it would be nice to use the type system to make sure `STATES` is a `const` etc.

**Comment by tschuy** [comment]
@iankronquist you reviewed a single commit. I readded the endpoint (recoded it) in a later commit.

**Comment by tschuy** [comment]
Specifically https://github.com/osuosl/standupbot/commit/165d3f28a69cb227006cb080e8f1e0cf1f8bed91

**Comment by iankronquist** [comment]
God damnit. It's way too easy to wind up looking at a single commit and think you're looking at the whole PR.

File: standupbot.js (hunk 1 above)
Comment (iankronquist) [inline standupbot.js]: This is still a problem.

File: package.json (hunk 2 above)
Comment (tschuy) [inline package.json:32]: @borcean was going to go through `standupbot` and clean up its lint as good js practice. I'm trying to write good js, but I'm going to leave adding `"use strict";` to him since that's a good thing to know about.

File: standupbot.js (hunk 1 above)
Comment (tschuy) [inline standupbot.js]: and this is also an HTML endpoint, so returning json might not be the best idea :)

**Comment by tschuy** [comment]
For now I'd prefer to get this merged, and handle 500 and 404 handling in #17. I've added a temporary database failure return.

**Comment by iankronquist** [comment]
Sweet. +1.

**Comment by pop** [comment]
The app runs, the tests pass (not that that means anything), and knex is muy bueno.
Plus One
+1
